# HG changeset patch
# User adinn
# Date 1517459341 0
#      Thu Feb 01 04:29:01 2018 +0000
# Node ID fb479b2ec08dc13d45b22180bc48807d6796b857
# Parent  a88da7093444356be05cc171fa99aae9abb8663c
PR3519: Fix functions with missing return value.
Reviewed-by: andrew

diff --git a/src/cpu/aarch64/vm/assembler_aarch64.hpp b/src/cpu/aarch64/vm/assembler_aarch64.hpp
--- openjdk/hotspot/src/cpu/aarch64/vm/assembler_aarch64.hpp
+++ openjdk/hotspot/src/cpu/aarch64/vm/assembler_aarch64.hpp
@@ -2529,6 +2529,8 @@
   virtual RegisterOrConstant delayed_value_impl(intptr_t* delayed_value_addr,
                                                 Register tmp,
                                                 int offset) {
+    ShouldNotCallThis();
+    return RegisterOrConstant();
   }
 
   // Stack overflow checking
diff --git a/src/cpu/aarch64/vm/c1_LIRAssembler_aarch64.cpp b/src/cpu/aarch64/vm/c1_LIRAssembler_aarch64.cpp
--- openjdk/hotspot/src/cpu/aarch64/vm/c1_LIRAssembler_aarch64.cpp
+++ openjdk/hotspot/src/cpu/aarch64/vm/c1_LIRAssembler_aarch64.cpp
@@ -209,6 +209,7 @@
 
 Address LIR_Assembler::as_Address_hi(LIR_Address* addr) {
   ShouldNotReachHere();
+  return Address();
 }
 
 Address LIR_Assembler::as_Address(LIR_Address* addr) {
diff --git a/src/cpu/aarch64/vm/macroAssembler_aarch64.hpp b/src/cpu/aarch64/vm/macroAssembler_aarch64.hpp
--- openjdk/hotspot/src/cpu/aarch64/vm/macroAssembler_aarch64.hpp
+++ openjdk/hotspot/src/cpu/aarch64/vm/macroAssembler_aarch64.hpp
@@ -434,8 +434,8 @@
   void mov(Register dst, Address a);
 
 public:
-  int push(RegSet regs, Register stack) { if (regs.bits()) push(regs.bits(), stack); }
-  int pop(RegSet regs, Register stack) { if (regs.bits()) pop(regs.bits(), stack); }
+  void push(RegSet regs, Register stack) { if (regs.bits()) push(regs.bits(), stack); }
+  void pop(RegSet regs, Register stack) { if (regs.bits()) pop(regs.bits(), stack); }
 
   // Push and pop everything that might be clobbered by a native
   // runtime call except rscratch1 and rscratch2.  (They are always
diff --git a/src/cpu/aarch64/vm/nativeInst_aarch64.hpp b/src/cpu/aarch64/vm/nativeInst_aarch64.hpp
--- openjdk/hotspot/src/cpu/aarch64/vm/nativeInst_aarch64.hpp
+++ openjdk/hotspot/src/cpu/aarch64/vm/nativeInst_aarch64.hpp
@@ -220,6 +220,7 @@
     else if (is_ldr_literal_at(instruction_address()))
       return(addr_at(4));
     assert(false, "Unknown instruction in NativeMovConstReg");
+    return NULL;
   }
 
   intptr_t data() const;
diff --git a/src/os_cpu/linux_aarch64/vm/os_linux_aarch64.cpp b/src/os_cpu/linux_aarch64/vm/os_linux_aarch64.cpp
--- openjdk/hotspot/src/os_cpu/linux_aarch64/vm/os_linux_aarch64.cpp
+++ openjdk/hotspot/src/os_cpu/linux_aarch64/vm/os_linux_aarch64.cpp
@@ -472,6 +472,7 @@
   err.report_and_die();
 
   ShouldNotReachHere();
+  return true; // Mute compiler
 }
 
 void os::Linux::init_thread_fpu_state(void) {
